packages/headless/src/features/analytics/analytics-utils.ts

```diff
@@ -745,3 +823,14 @@ export const makeCommerceAnalyticsAction = <
     }
   );
 };
+
+function logNextEvent<PayloadType extends RelayPayload = RelayPayload>(
+  emitEvent: (
+    type: string,
+    payload: RelayPayload
+  ) => Promise<void | ValidationResponse>,
```

Comment:
Would it be possible to remove `ValidationResponse` from the possible return types? While it's supported now, this is something we plan on changing once the Explorer debugging tool is available. Validation will be sent to the tool rather than returned by the `emit` function.

Since `emit` will soon use beacon api under the hood, it will become fire-and-forget removing the need to await a promise. Maybe the return type here can simply be `void`? If `tsc` complain though, we can do this in a future PR,

Reply:
- [ ] Todo: Coerce to void